src/main/java/org/jenkinsci/plugins/saml/SamlAuthenticationToken.java

```diff
@@ -45,4 +45,19 @@ public String getCredentials() {
         return "SAML does not use passwords";
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        if(obj instanceof SamlAuthenticationToken){
+            return userDetails.equals(obj);
+        }
+        else {
+            return false;
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return userDetails.hashCode();
```

Comment:
weak implementation uses the Object.hashCode(), is mostly the same than not implement it